components/DetectAnimatedGif.coffee

```diff
@@ -22,7 +22,27 @@ exports.getComponent = ->
     async: true
     forwardGroups: true
   , (buffer, groups, out, callback) ->
-    out.send animated buffer
-    do callback
+    if Buffer.isBuffer buffer
+      out.send animated buffer
+      do callback
+      return
+    else if typeof buffer is 'string'
+      isAnimated = false
+      fs.createReadStream buffer
+        .pipe animated()
+        .once 'animated', ->
+          isAnimated = true
+          out.send true
+          do callback
+          return
+        .on 'finish', ->
+          unless isAnimated
```

Comment:
Ah you're right, with last version of `animated-gif-detector` it sends `animated` event and `finish`, so we just need to deal with that on `finish`, right, right.